--- a/nova/virt/libvirt/driver.py
+++ b/nova/virt/libvirt/driver.py
@@ -8098,11 +8098,13 @@ class LibvirtDriver(driver.ComputeDriver):
 
         data['pci_passthrough_devices'] = self._get_pci_passthrough_devices()
 
-        numa_topology = self._get_host_numa_topology()
-        if numa_topology:
-            data['numa_topology'] = numa_topology._to_json()
-        else:
-            data['numa_topology'] = None
+        # (erikberg) disable numa_topology
+        #numa_topology = self._get_host_numa_topology()
+        #if numa_topology:
+        #    data['numa_topology'] = numa_topology._to_json()
+        #else:
+        #    data['numa_topology'] = None
+        data['numa_topology'] = None
 
         return data
 
