--- launch-instance-model.service.js    2021-02-24 15:51:01.525689566 +0100
+++ launch-instance-model.service.js    2021-02-24 21:41:29.248678206 +0100
@@ -203,7 +203,7 @@
         source: [],
         create_volume_default: true,
         // REQUIRED for JS logic
-        hide_create_volume: false,
+        hide_create_volume: true,
         vol_create: false,
         // May be null
         vol_device_name: 'vda',
@@ -298,9 +298,6 @@
         // Append "_default" to distinguish from the 'vol_create' item
         model.newInstanceSpec.create_volume_default = defaults.create_volume;
       }
-      if ('hide_create_volume' in defaults) {
-        model.newInstanceSpec.hide_create_volume = defaults.hide_create_volume;
-      }
     }

     /**
