diff --git a/openstack_dashboard/dashboards/project/instances/tables.py b/openstack_dashboard/dashboards/project/instances/tables.py
index d385d7995..3323bfc70 100644
--- a/openstack_dashboard/dashboards/project/instances/tables.py
+++ b/openstack_dashboard/dashboards/project/instances/tables.py
@@ -1024,7 +1024,7 @@ def get_flavor(instance):
             "flavor_id": instance.full_flavor.id
         }
         return template.loader.render_to_string(template_name, context)
-    return _("Not available")
+    return _("Outdated")
 
 
 def get_keyname(instance):
