--- /usr/lib/python3.6/site-packages/nova/virt/libvirt/config.py_ORIG	2024-06-14 16:09:58.009451408 +0200
+++ /usr/lib/python3.6/site-packages/nova/virt/libvirt/config.py	2024-06-14 16:11:11.143179641 +0200
@@ -3165,6 +3165,7 @@
                                         root_name="capability", **kwargs)
         self.type = None
         self.iommu_group = None
+        self.uuid = None
 
     def parse_dom(self, xmldoc):
         super(LibvirtConfigNodeDeviceMdevInformation,
@@ -3174,6 +3175,8 @@
                 self.type = c.get('id')
             if c.tag == "iommuGroup":
                 self.iommu_group = int(c.get('number'))
+            if c.tag == "uuid":
+                self.uuid = c.text
 
 
 class LibvirtConfigGuestRng(LibvirtConfigGuestDevice):
