--- /usr/lib/python3.6/site-packages/nova/virt/libvirt/config.py_ORIG   2022-04-07 16:21:01.000000000 +0200
+++ /usr/lib/python3.6/site-packages/nova/virt/libvirt/config.py        2024-06-14 14:18:01.842169813 +0200
@@ -3046,6 +3046,8 @@
         for c in xmldoc:
             if c.tag == "name":
                 self.name = c.text
+                if len(self.name) > 41:
+                    self.name = self.name[:41]
             elif c.tag == "parent":
                 self.parent = c.text
             elif c.tag == "capability" and c.get("type") in ['pci', 'net']:
